docs/draft_production_docs/oadp-self-service.md

```diff
@@ -0,0 +1,484 @@
+
+* The OADP DPA has been configured to support self-service
+* The cluster administrator has created the users 
+  * account 
+  * namespace
+  * namespace privileges, e.g. namespace admin.
+  * optionally the cluster administrator can create a NABSL for the user.
+
+### OpenShift self-service required permissions:
+
+Ensure users have appropriate permissions in its namespace.  Users must have editor roles for the following objects in their namespace.
+  * nonadminbackups.oadp.openshift.io                                
+  * nonadminbackupstoragelocations.oadp.openshift.io                
+  * nonadminrestores.oadp.openshift.io                              
+
+  For example
+  ```yaml
+    # config/rbac/nonadminbackup_editor_role.yaml
+    - apiGroups:
+        - oadp.openshift.io
+      resources:
+        - nonadminbackups
+        - nonadminrestores
+        - nonadminbackupstoragelocations
+      verbs:
+        - create
+        - delete
+        - get
+        - list
+        - patch
+        - update
+        - watch
+    - apiGroups:
+        - oadp.openshift.io
+      resources:
+        - nonadminbackups/status
+      verbs:
+        - get
+    # config/rbac/nonadminrestore_editor_role.yaml
+    - apiGroups:
+      - oadp.openshift.io
+      resources:
+      - nonadminrestores
+      verbs:
+      - create
+      - delete
+      - get
+      - list
+      - patch
+      - update
+      - watch
+    - apiGroups:
+      - oadp.openshift.io
+      resources:
+      - nonadminrestores/status
+      verbs:
+      - get
+  ```
+
+  **note** users will not be able to edit the NABSL's. Users will only be able to create NABSL's.  
+
+## Self-Service workflow
+
+Non Cluster Administrators can utilize OADP self-service by creating NonAdminBackup (NAB) and NonAdminRestore (NAR) objects in the namespace to be backed up or restored.  A NonAdminBackup is an OpenShift custom resource that securely facilitates the creation, status and lifecycle of a Velero Backup custom resource.  
+
+```mermaid
+sequenceDiagram
+    participant User
+    participant NAB as NonAdminBackup
+    participant NAC as NonAdminController
+    participant VB as Velero Backup
+
+    User->>NAB: Creates NonAdminBackup CR
+    NAB->>NAC: Detected by controller
+    NAC-->>NAB: Validates backup request
+    NAC->>VB: Creates Velero Backup CR
+    VB-->>NAB: Status updates
+    NAB-->>User: View backup status
+```
+
+![nab-backup-workflow](https://hackmd.io/_uploads/BJz4bEbKyx.jpg)
+
+For the most part one can think of a NonAdminBackup and a Velero Backup in very much the same way.  Both objects specify a velero backup and how the backup should be executed.  There are a few differences to keep in mind when creating a NonAdminBackup.
+
+1. The NonAdminBackup creates the Velero Backup CR instance in a secure way that limits the users access.
```

Comment:
The credentials and NaBSL referenced in the NonAdminBackup, or the name referenced in the NAR, are local namespace reference names or local NAB names, rather than Velero CR names.

Reply:
I think here its fine to leave that out, we are only talking about high level things, we can mark this up in the spec docs of NAC APIs